server/autolinkapp/app.go

```diff
@@ -0,0 +1,172 @@
+package autolinkapp
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+	"github.com/mattermost/mattermost-plugin-autolink/server/autolink"
+)
+
+type Store interface {
+	GetLinks() []autolink.Autolink
+	SaveLinks([]autolink.Autolink) error
+}
+
+type Service interface {
+	GetPluginURL() string
+	IsAuthorizedAdmin(userID string) (bool, error)
+}
+
+type app struct {
+	store   Store
+	service Service
+}
+
+const (
+	appID          = "com.mattermost.autolink"
+	appDisplayName = "Autolink"
+
+	autolinkCommand = "autolink-app"
+
+	appRoute      = "/app/v1"
+	manifestRoute = "/manifest"
+	bindingsRoute = "/bindings"
+	iconRoute     = "/public/icon.png" // Served from the plugin's public folder
+
+	editLinksRoute        = "/edit-links"
+	editLinksCommandRoute = "/edit-links-command/submit"
+)
+
+func RegisterHandler(root *mux.Router, store Store, service Service) {
+	a := &app{
+		store:   store,
+		service: service,
+	}
+
+	appRouter := root.PathPrefix(appRoute).Subrouter()
+	appRouter.HandleFunc(manifestRoute, a.handleManifest).Methods(http.MethodGet)
+
+	adminRoutes := appRouter.PathPrefix("").Subrouter()
+	adminRoutes.Use(a.adminRequired)
+
+	adminRoutes.HandleFunc(bindingsRoute, a.handleBindings).Methods(http.MethodPost)
+	adminRoutes.HandleFunc(editLinksRoute+"/form", a.handleFormFetch).Methods(http.MethodPost)
+	adminRoutes.HandleFunc(editLinksRoute+"/submit", a.handleFormSubmit).Methods(http.MethodPost)
+	adminRoutes.HandleFunc(editLinksCommandRoute+"/submit", a.handleCommandSubmit).Methods(http.MethodPost)
```

Comment:
Is this really `/edit-links-command/submit/submit`?

Reply:
Hmm that was happening before and I thought I fixed that. Not sure how/why it's working